src/main/java/com/aws/greengrass/clientdevices/auth/iot/IotAuthClient.java

```diff
@@ -145,8 +149,8 @@ public boolean isThingAttachedToCertificate(Thing thing, String certificateId) {
                     VerifyClientDeviceIoTCertificateAssociationRequest.builder()
                             .clientDeviceThingName(thing.getThingName())
                             .clientDeviceCertificateId(certificateId).build();
-            try {
-                clientFactory.getGreengrassV2DataClient().verifyClientDeviceIoTCertificateAssociation(request);
```

Comment:
handle the `DeviceConfigurationException` nicely in these places. Right now it will fallthrough to the log about checking permissions, which doesn't apply to this error.

Reply:
It falls through the catch-all block that also logs the cause (stacktrace) of this exception with a generic failure message.

Reply:
Let me add a catch block `DeviceConfigurationException` to make the cause more clear.